google/cloud/internal/url_encode_test.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/url_encode.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using google::cloud::internal::UrlEncode;
+using google::cloud::internal::UrlDecode;
```

Comment:
I'm not sure why you needed that, given that we're already within `google::cloud::internal`.

Reply:
I was getting 
```
Step #3: ../google/cloud/internal/url_encode_test.cc: In member function 'virtual void google::cloud::internal::v2_17::{anonymous}::UrlDecode_MultipleReplacements_Test::TestBody()':
Step #3: ../google/cloud/internal/url_encode_test.cc:54:17: error: 'UrlDecode' was not declared in this scope
Step #3:    auto result = UrlDecode(encoded_string);
Step #3:                  ^~~~~~~~~
Step #3: ../google/cloud/internal/url_encode_test.cc:54:17: note: suggested alternative:
Step #3: In file included from ../google/cloud/internal/url_encode_test.cc:15:0:
Step #3: ../google/cloud/internal/url_encode.h:29:13: note:   'google::cloud::v2_17::internal::UrlDecode'
Step #3:  std::string UrlDecode(absl::string_view value);
Step #3:              ^~~~~~~~~
``` 
On the cmake build

Reply:
It looks like the `test.cc` does not agree with the `h` and `.cc` files on where the inline namespace should occur.

Reply:
Ugh, that makes sense. Fixed. Thanks for catching it.